examples/ui/ui_material.rs

```diff
@@ -23,16 +23,16 @@ fn setup(
     commands.spawn(Camera2d);
 
     commands
-        .spawn(NodeBundle {
-            style: Style {
+        .spawn((
+            Node::default(),
+            Style {
                 width: Val::Percent(100.0),
                 height: Val::Percent(100.0),
                 align_items: AlignItems::Center,
                 justify_content: JustifyContent::Center,
                 ..default()
             },
-            ..default()
-        })
+        ))
         .with_children(|parent| {
             let banner_scale_factor = 0.5;
             parent.spawn(MaterialNodeBundle {
```

Comment:
Is there a reason we need BackgroundColor to break UI material extraction? Can we just remove that filter? Seems to work for me.

Reply:
Ah it does seem like it would still be captured by `extract_uinode_background_colors`. Munching on this now.

Reply:
This still works by default because BackgroundColor defaults to transparent, which results in material node backgrounds being skipped by default.

This does slightly break if set the background color to a visible color (because the material and the background z-fight with each other). I think ideally we make background color play nicely with this case.

Reply:
I've fixed this in a satisfying way that also paves the way for other such "multi-layered extraction" cases. I've also ported MaterialNodeBundle to required components and renamed `UiMaterialHandle<M>` to `MaterialNode<M>`. It is now the "driving concept" and requires `Node`.

Reply:
Polishing it up now. I'll push momentarily.

Reply:
Just pushed this.